urlb57eb64056_jacadenac_ISI_Othello_UN_tgz-pJ8-othello_OthelloJ8/src/othello/UIJuego.java

```diff
@@ -0,0 +1,427 @@
+			return 12;
+		} else if (tableroUI.obtenerTamTableroButton()[2].isSelected()) {
+			return 14;
+		}else {
+			return 16;
+		}
+	}
+	
+	public void definirTipoTablero(int tamanio, TipoTablero tipoTablero){
+		tableroUI.dispose();
+		
+		if ( tipoTablero == TipoTablero.OCTOGONAL ) {
+			JFileChooser fileChooser = new JFileChooser();
+			String nombreFichero = "Octogonal10x10.oth";
+			if (tamanio == 12) {
+				nombreFichero = "Octogonal12x12.oth";
+			} else if (tamanio == 14) {
+				nombreFichero = "Octogonal14x14.oth";
+			} else if (tamanio == 16) {
+				nombreFichero = "Octogonal16x16.oth";
+			}
+			File directorio = new File("setup/boards");
+			File fichero = new File(directorio.getPath(),nombreFichero);
+			fileChooser.setCurrentDirectory(fichero);		
+			fileChooser.setFileFilter(miFiltro);
+			try {
+				leerArchivo(fichero, tipoTablero);
+				tableroUI = new TableroUI(Tablero.TABLERO_LARGO, Tablero.TABLERO_ANCHO, controlador.obtenerTableroLogica(), tipoTablero);
+				iniciarTableroUI();
+			} catch (FileNotFoundException e1) {
+				e1.printStackTrace();
+			}
+		}
+		
+		if ( tipoTablero == TipoTablero.CLASICO || tipoTablero == null) {
+			controlador.inicializar(tamanio, tamanio, tipoTablero);
+			tableroUI = new TableroUI(Tablero.TABLERO_LARGO, Tablero.TABLERO_ANCHO, controlador.obtenerTableroLogica(), tipoTablero);
+			iniciarTableroUI();
+		}
+		
+		if ( tipoTablero == TipoTablero.PERSONALIZADO ){
+			controlador.inicializar(tamanio, tamanio, tipoTablero);
+			tableroUI = new TableroUI(Tablero.TABLERO_LARGO, Tablero.TABLERO_ANCHO, controlador.obtenerTableroLogica(), tipoTablero);
+			iniciarTableroUI();
+			estaModificandoTablero = true;
+			tableroUI.mostrarBarraBoton();	
+		}
+		
+		run();
+	}
+	
+	private void escribirArchivo(File fichero) {
+		PrintWriter writer;
+		try {
+			int casillasJugablesIniciales=0;
+			writer = new PrintWriter(fichero);
+			Map<Point, EstadoCasilla> tableroActual = controlador.obtenerTableroLogica().obtenerTablero();
+			writer.println(TableroUI.largoTablero + "," + TableroUI.anchoTablero + "," + controlador.jugadorActual().color()
+					+ "," + obtenerTipoTableroElegido());
+			for(Point point : tableroActual.keySet()) {
+				writer.println(point.x + "," + point.y + "," + tableroActual.get(point).toString());
+				if(tableroActual.get(point) != EstadoCasilla.WALL){
+					casillasJugablesIniciales++;
+				}
+			}
+			writer.println("-"+casillasJugablesIniciales);
+			writer.close();
+		} catch (FileNotFoundException e) {
+			e.printStackTrace();
+		}
+	}
+	
+	private void leerArchivo(final File fichero, TipoTablero tipoTablero) throws FileNotFoundException {
+		BufferedReader reader;
+		int largoTablero=14;
+		int anchoTablero=14;
+		Map<Point, EstadoCasilla> crearTablero = new HashMap<Point, EstadoCasilla>();
+		EstadoCasilla colorJugador = EstadoCasilla.BLACK;
+		boolean esTamańo=true;
+		
+		try {
+		    int casillasJugablesIniciales=0;
+			reader = new BufferedReader(new FileReader(fichero));
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

**Problem**
This line of code contains a resource that might not be closed properly. This can cause a resource leak that slows down or crashes your system.

**Fix**
Consider closing the following resource: *reader*. The resource is referenced at lines 392, 393, 414. The resource is closed at line 420. There are other execution paths that do not contain closure statements, for example, when BufferedReader.readLine throws an exception. Ensure *reader* is closed in a try-finally block or declared in a try-with-resources block.

**More info**
[View resource management guidelines at oracle.com](https://www.oracle.com/technetwork/articles/java/trywithresources-401775.html) (external link).